src/blocks/weather/nws.rs

```diff
@@ -0,0 +1,381 @@
+        match icon {
+            WeatherIcon::Clear { .. } => "Clear",
+            WeatherIcon::Clouds { .. } => "Clouds",
+            WeatherIcon::Fog { .. } => "Fog",
+            WeatherIcon::Thunder { .. } => "Thunder",
+            WeatherIcon::Rain { .. } => "Rain",
+            WeatherIcon::Snow => "Snow",
+            WeatherIcon::Default => "Unknown",
+        }
+        .to_string()
+    }
+
+    fn wind_kmh(&self) -> f64 {
+        if self.wind_speed.unit_code.ends_with("km_h-1") {
+            self.wind_speed.value
+        } else {
+            self.wind_speed.value * MPH_TO_KPH
+        }
+    }
+
+    fn apparent_temp(&self) -> f64 {
+        let temp = if self.temperature.unit_code.ends_with("degC") {
+            self.temperature.value
+        } else {
+            (self.temperature.value - 32.0) * 5.0 / 9.0
+        };
+        let humidity = self.relative_humidity.value;
+        let wind_speed = self.wind_kmh();
+        australian_apparent_temp(temp, humidity, wind_speed)
+    }
+
+    fn to_moment(&self) -> WeatherMoment {
+        let icon = short_forecast_to_icon(&self.short_forecast, !self.is_daytime);
+        let weather = Self::icon_to_word(icon);
+        WeatherMoment {
+            icon,
+            weather,
+            weather_verbose: self.short_forecast.clone(),
+            temp: self.temperature.value,
+            apparent: self.apparent_temp(),
+            humidity: self.relative_humidity.value,
+            wind: self.wind_speed.value,
+            wind_kmh: self.wind_kmh(),
+            wind_direction: self.wind_direction(),
+        }
+    }
+
+    fn to_aggregate(&self) -> ForecastAggregate {
+        ForecastAggregate {
+            temp: self.temperature.value,
+            apparent: self.apparent_temp(),
+            humidity: self.relative_humidity.value,
+            wind: self.wind_speed.value,
+            wind_kmh: self.wind_kmh(),
+            wind_direction: self.wind_direction(),
+        }
+    }
+}
+
+fn combine_forecasts(data: &[ForecastAggregate], fin: WeatherMoment) -> Forecast {
+    let mut temp = 0.0;
+    let mut apparent = 0.0;
+    let mut humidity = 0.0;
+    let mut wind_north = 0.0;
+    let mut wind_east = 0.0;
+    let mut wind_kmh_north = 0.0;
+    let mut wind_kmh_east = 0.0;
+    let mut max = ForecastAggregate {
+        temp: f64::MIN,
+        apparent: f64::MIN,
+        humidity: f64::MIN,
+        wind: f64::MIN,
+        wind_kmh: f64::MIN,
+        wind_direction: None,
+    };
+    let mut min = ForecastAggregate {
+        temp: f64::MAX,
+        apparent: f64::MAX,
+        humidity: f64::MAX,
+        wind: f64::MAX,
+        wind_kmh: f64::MAX,
+        wind_direction: None,
+    };
+    for val in data {
+        // Summations for averaging
+        temp += val.temp;
+        apparent += val.apparent;
+        humidity += val.humidity;
+        let (sin, cos) = val
+            .wind_direction
+            .unwrap_or_default()
```

Comment:
Ah and instead of treating None as north just don't use that reading to contribute to the average

Reply:
Got it, I updated the wind average calculation to match up with what the `average_wind` function does in the `weather.rs` code. For what it's worth, I do still want to keep this `combine_forecasts` function even though there's stuff like `average_wind(...)`, as it seems like a general function that merges together `ForecastAggregate`s would be more reusable down the line? Once I stop having so many little mistakes in it, anyway.

Reply:
Yes, I was actually thinking of (in a later PR) moving your new function to combine the forecast up the weather module and using it for the other weather backends.

I was thinking too about how we can make is so that we don't have to precompute the apparent weather and wind kmh for each forecast used to create the average/min/max

Reply:
So, I was wondering about what the approach should be for measurement units as well. Wind is the only one that has two potentially different units at the same time, and confusingly to me, there's actually *three* different units in use in the code: m/s, km/h, and mph. This means I have at least one more tweak to make - my code calls the `australian_apparent_temp` function with a value in km/h, when it should actually be in m/s.

Apparent temperature also results in a measurement in degrees Celsius, and I should probably be converting that back into a temperature in degrees Fahrenheit *if* that's the chosen measurement unit. OpenWeatherMap seems to keep the feels-like temperature in Fahrenheit, so I'll need to do that to be consistent.

I'm not sure it matters all that much if I compute the wind speed twice, as it's a pretty infrequent calculation, but it might be good if there weren't two wind speed measurements in `WeatherAggregate` and `WeatherMoment` to begin with. My proposed change (probably not in this pull request) would be to add a `units: UnitSystem` field to `WeatherResult`, ditch the `wind_kmh` fields, and patch the display code to check the unit system and convert to km/h when someone uses the `wind_kmh` format keys.

Reply:
Whoops, I partially misunderstood the issue. It seems everyone uses m/s for wind when in metric units, and km/h is an alternate view of that. I still think the fix of patching the display code and lifting `wind_kmh` out of the weather data structures would work though.

Reply:
Does met.no have any support for imperial units at all? I thought it was fixed to being metric only.

Reply:
Sorry, right you are 😅